docs/guide/writing-tasks.md

```diff
@@ -126,16 +126,19 @@ test_task:
 ```
 
 A `fingerprint_script` is an optional field that can specify a script that will be executed and console output of which
-will be used as a fingerprint for the given task. By default task name is used as a fingerprint value.
+will be used as a fingerprint for the given task. By default the task name is used as a fingerprint value.
 
-`populate_script` is an optional field that can specify a script that will be executed to populate the cache. 
+`populate_script` is an optional field that can specify a script that will be executed to populate the cache.
 `populate_script` should create the `folder` if it doesn't exist before the `cache` instruction.
+If your dependencies are updated often, we suggest putting the populate script after the cache block so new versions of dependencies are used.
```

Comment:
It's generally a bad design if your dependencies can change without changing `fingerprint_script`'s output. I'd say that if your dependencies are updated often, people should pay extra attention to the fingerprint.

Reply:
True.  I'll remove this.